kin/account.py

```diff
@@ -0,0 +1,347 @@
+"""Contains the KinAccount and AccountStatus classes."""
+
+from functools import partial
+
+from enum import Enum
+from stellar_base.asset import Asset
+
+from .blockchain.keypair import Keypair
+from .blockchain.horizon import Horizon
+from .blockchain.builder import Builder
+from .blockchain.channel_manager import ChannelManager
+from . import errors as KinErrors
+from .transactions import Transaction
+from .blockchain.errors import TransactionResultCode, HorizonErrorType , HorizonError
+from .config import MIN_ACCOUNT_BALANCE, SDK_USER_AGENT
+from .blockchain.utils import is_valid_secret_key, is_valid_address
+
+import logging
+logger = logging.getLogger(__name__)
+
+
+class KinAccount:
+    """Account class to perform authenticated actions on the blockchain"""
+    def __init__(self, seed, client, channels, channel_secret_keys, create_channels):
```

Comment:
is it ok to make the init blocking? is there any best practice regarding this? meaning does SDK user should know what is blocking and what's not?

Reply:
Everything is expected to be blocking unless running in threads (like the monitoring method)